### PR TITLE
Add Caldav configuration flow documentation

### PR DESCRIPTION
**Pull request by allenporter**
## Proposed change

Add Caldav configuration flow documentation, and refresh much of the stale documentation to nudge users towards UI and triggers rather than yaml, offsets, and entity attributes. Some of this is copied from google calendar.

The motivation is to prepare for supporting `VTODO` objects from CalDAV to enable things like iCloud To-do lists.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/103215
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/caldav.markdown [hunk 1]
```diff
@@ -11,25 +11,24 @@ ha_platforms:
 ha_integration_type: integration
 ---
 
-The `caldav` platform allows you to connect to your WebDAV calendar and generate binary sensors. A different sensor will be created for each individual calendar, or you can specify custom calendars which match a criteria you define (more on that below). These sensors will be `on` if you have an on going event in that calendar or `off` if the event is later in time, or if there is no event at all. The WebDAV calendar get updated roughly every 15 minutes.
+The CalDAV integration allows you to connect your WebDAV calendar to Home Assistant.
+Each calenddar is a [calendar](/integrations/calendar) entity which can be used
```
Comment (MartinHjelmare) [inline source/_integrations/caldav.markdown]:
```suggestion
Each calendar is a [calendar](/integrations/calendar) entity which can be used
```

File: source/_integrations/caldav.markdown [hunk 2]
```diff
@@ -11,25 +11,24 @@ ha_platforms:
...
-{% details "Notes for Home Assistant Core Installations" %}
+This integration was tested against [iCloud Calendar](https://www.icloud.com/calendar/), [Nextcloud](https://nextcloud.com/), [Owncloud](https://owncloud.org/), [Baikal](https://sabre.io/baikal/). Any integration complying with the RFC4791 should work.
 
-You might need some additional system packages to compile the Python CalDAV library. On a Debian based system, install them by:
+## Manual Configuration
 
-```bash
-sudo apt-get install libxml2-dev libxslt1-dev zlib1g-dev
-```
+You may also manually WebDAV calendar in Home Assistant by adding the following section to your `configuration.yaml` file:
```
Comment (MartinHjelmare) [inline source/_integrations/caldav.markdown]:
```suggestion
You may also manually add a WebDAV calendar in Home Assistant by adding the following section to your `configuration.yaml` file:
```

File: source/_integrations/caldav.markdown [hunk 3]
```diff
@@ -67,9 +66,9 @@ calendar:
     url: https://caldav.icloud.com
 ```
 
-This example will generate default binary sensors for each calendar you have in your account. Those calendars will be `on` when there is an ongoing event and `off` if not. Events that last a whole day are ignored in those calendars. You have to setup custom calendars in order to take them into account or for advanced event filtering.
+This example will generate default binary sensors for each calendar you have in your account. Those calendars will be `on` when there is an ongoing event and `off` if not. Events that last a whole day are ignored in those calendars.
```
Comment (MartinHjelmare) [inline source/_integrations/caldav.markdown]: Talking about binary sensors is confusing when it's calendar entities that are meant.

**Review by MartinHjelmare — CHANGES_REQUESTED** [review]

File: source/_integrations/caldav.markdown (hunk 3 above)
Comment (allenporter) [reply source/_integrations/caldav.markdown]: Updated. I tried to remove as much unnecessary/incorrect text as possible.

**Review by MartinHjelmare — APPROVED** [review]
Thanks!

**Pull request merged**